extractuser.go

```
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"crypto/rsa"
+
+	"golang.org/x/net/context"
+
+	"net/http"
+
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+)
+
+// ConfigureExtractUser is a configuration for middleware,
+// Accepts PublicKey and PrivateKey to create a token manager
+// Using token manager, it is responsible for extracting the token from context if present
+// Update the context with uuid found in token
+func ConfigureExtractUser(publicKey *rsa.PublicKey, privateKey *rsa.PrivateKey) (ExtractUser goa.Middleware) {
+	manager := token.NewManager(publicKey, privateKey)
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+			// for now ignore the error becasue still test for logged in user is not done.
+			uuid, _ := manager.Locate(ctx)
+			ctxWithUser := context.WithValue(ctx, "uuid", uuid.String())
```

1. Won't this crash when there is not user?
2. Why convert uuid to string?

---

1. yes, you are right, error is ignored temporarily. And when there is no token found, "Locate" returns `uuid.UUID{}` which is `"000-00...all zeros"` 
2. As of now, it is simple to keep string because for remote-work-items creator/user is not created before actual importing. So anything can go in it for now :)